### PR TITLE
fix: check both success and termination for the dependency predicate

### PR DESCRIPTION
**Pull request by rapsealk**
Several cases have been reported that sessions with dependency are not executed in sequence; a session runs even its precedent session has not been terminated yet.

This PR resolves this by checking if a dependent session has both succeeded and terminated.

**Checklist:** (if applicable)

- [x] Milestone metadata specifying the target backport version
- [ ] Mention to the original issue
- [ ] Documentation
- [ ] API server-client counterparts (e.g., manager API -> client SDK)
- [ ] Test case(s) to demonstrate the difference of before/after

### CODE REVIEW

**Review by fregataa — COMMENTED** [review]
This is good because it reduces confusion between session dependencies for users, but I think there should be more discussion about dependency check in general before we merge this.
How do you think of this? @achimnol

File: src/ai/backend/manager/scheduler/predicates.py [hunk 1]
```diff
@@ -128,7 +130,7 @@ async def check_dependencies(
     rows = result.fetchall()
     pending_dependencies = []
     for row in rows:
-        if row.result != SessionResult.SUCCESS:
+        if row.result != SessionResult.SUCCESS or row.status != SessionStatus.TERMINATED:
```
Comment (kyujin-cho) [inline src/ai/backend/manager/scheduler/predicates.py:133]: Is it okay not to cover `CANCELLED` session status here?

**Review by kyujin-cho — CHANGES_REQUESTED** [review]

File: src/ai/backend/manager/scheduler/predicates.py (hunk 1 above)
Comment (rapsealk) [reply src/ai/backend/manager/scheduler/predicates.py:133]: I think those which depend on `CANCELLED` sessions are not supposed to be started.

Comment (rapsealk) [reply src/ai/backend/manager/scheduler/predicates.py:133]: They rather be notified and `CANCELLED` either?

Comment (kyujin-cho) [reply src/ai/backend/manager/scheduler/predicates.py:133]: Let's keep this as is for now and create a patch if same issue occurs for `CANCELLED` sessions.

**Review by kyujin-cho — APPROVED** [review]

**Pull request merged**